The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5388070033/jobs/9780229217) [(plain)](https://github.com/rust-lang/rust/commit/142bb39f26b519bd0d930812aff09ca95aba1d99/checks/14583666838/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [ui] tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs stdout ----
diff of stderr:

16 #0 [early_lint_checks] perform lints prior to macro expansion
17 #1 [hir_crate] getting the crate HIR
+ stack backtrace:
+ stack backtrace:
+    0:     0x7f64b76a4231 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h6dea4e7324a59a82
+    1:     0x7f64b770d258 - core::fmt::write::hfd96dfa72d0a2725
+    2:     0x7f64b7698711 - std::io::Write::write_fmt::h2e57173eeb62de35
+    3:     0x7f64b76a4041 - std::sys_common::backtrace::print::he2c4eabdf3f324b5
+    4:     0x7f64b76a71da - std::panicking::default_hook::{{closure}}::h5ac93fa12602a939
+    5:     0x7f64b76a6ec3 - std::panicking::default_hook::h744089070e358704
+    6:     0x7f64b83f580b - rustc_driver_impl[83116b0749ee6832]::install_ice_hook::{closure#0}
+    7:     0x7f64b76a791e - std::panicking::rust_panic_with_hook::hfa2b2d0711ad450a
+    8:     0x7f64b76a7679 - std::panicking::begin_panic_handler::{{closure}}::h5f4d8dedc2462c74
+    9:     0x7f64b76a46f6 - std::sys_common::backtrace::__rust_end_short_backtrace::h03e91b6dad2b248d
+   10:     0x7f64b76a7387 - rust_begin_unwind
+   11:     0x7f64b765b0d3 - core::panicking::panic_fmt::h469794c234f7c718
+   12:     0x7f64baaf75ef - <rustc_lint[63ba0b95152488e9]::levels::BuilderPush as core[ef4d13f605d7c29c]::ops::drop::Drop>::drop
+   13:     0x7f64b84958b5 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_expr
+   14:     0x7f64b8496980 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_stmt
+   15:     0x7f64b844fbdc - rustc_ast[5f32a37bf33b9536]::visit::walk_block::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
+   16:     0x7f64b84971c5 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_block
+   17:     0x7f64b849daa0 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_fn
+   18:     0x7f64b8456c9c - rustc_ast[5f32a37bf33b9536]::visit::walk_item::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
+   19:     0x7f64b8495bac - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_item
+   20:     0x7f64b844fd0c - rustc_ast[5f32a37bf33b9536]::visit::walk_crate::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
+   21:     0x7f64b8485de1 - rustc_lint[63ba0b95152488e9]::early::check_ast_node::<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass, (&rustc_ast[5f32a37bf33b9536]::ast::Crate, &[rustc_ast[5f32a37bf33b9536]::ast::Attribute])>
+   22:     0x7f64b84eb1eb - rustc_interface[d667fd10b95d5f1c]::passes::early_lint_checks
+   23:     0x7f64b9e38ce5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>
+   24:     0x7f64b9fa03c0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   25:     0x7f64ba11c68f - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   26:     0x7f64ba00da2d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::get_query_non_incr::__rust_end_short_backtrace
+   27:     0x7f64b9255930 - rustc_ast_lowering[2dcdb8df9e5d1db7]::lower_to_hir
+   28:     0x7f64b9e50015 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
+   29:     0x7f64b9f73240 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   30:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   31:     0x7f64ba00dafa - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::get_query_non_incr::__rust_end_short_backtrace
+   32:     0x7f64bb14d014 - <rustc_middle[f3ae9a0a99208368]::hir::provide::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
+   33:     0x7f64b9e502ae - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>
+   34:     0x7f64b9f040ac - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
+   35:     0x7f64ba1b4251 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   36:     0x7f64ba0ed246 - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::get_query_non_incr::__rust_end_short_backtrace
+   37:     0x7f64bb16e080 - rustc_middle[f3ae9a0a99208368]::query::plumbing::query_get_at::<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>>
+   38:     0x7f64bb1509db - <rustc_middle[f3ae9a0a99208368]::hir::map::Map>::get_module
+   39:     0x7f64bb1559ec - rustc_middle[f3ae9a0a99208368]::hir::map::hir_crate_items
+   40:     0x7f64b9e344b5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
+   41:     0x7f64b9f6fbb0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   42:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   43:     0x7f64ba0a0b6a - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::get_query_non_incr::__rust_end_short_backtrace
+   44:     0x7f64b96357b0 - rustc_passes[efdb855c787defeb]::hir_id_validator::check_crate
+   45:     0x7f64b84f04cf - rustc_interface[d667fd10b95d5f1c]::passes::analysis
+   46:     0x7f64b9e4ee05 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>
+   47:     0x7f64ba0ebbe0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   48:     0x7f64ba12135e - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   49:     0x7f64ba0a2b9d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
+   50:     0x7f64b83c2e4d - <rustc_middle[f3ae9a0a99208368]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   51:     0x7f64b83d3b08 - <rustc_interface[d667fd10b95d5f1c]::interface::Compiler>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}, core[ef4d13f605d7c29c]::result::Result<core[ef4d13f605d7c29c]::option::Option<rustc_interface[d667fd10b95d5f1c]::queries::Linker>, rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   52:     0x7f64b83a2d58 - rustc_span[90e41021240f79d4]::set_source_map::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
+   53:     0x7f64b8388d91 - <scoped_tls[c2bdb5f8b9cb5d8d]::ScopedKey<rustc_span[90e41021240f79d4]::SessionGlobals>>::set::<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   54:     0x7f64b841c926 - std[579a62aacf84a528]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   55:     0x7f64b8419bb6 - std[579a62aacf84a528]::panicking::try::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, core[ef4d13f605d7c29c]::panic::unwind_safe::AssertUnwindSafe<<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
+   56:     0x7f64b83a13af - <<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
+   57:     0x7f64b76b417d - std::sys::unix::thread::Thread::new::thread_start::h09e0297455ccbbb3
+   58:     0x7f64b744eb43 - <unknown>
+   59:     0x7f64b74e0a00 - <unknown>
+   60:                0x0 - <unknown>
+ error: the compiler unexpectedly panicked. this is a bug.
+ 
+ query stack during panic:
+ query stack during panic:
+ #0 [early_lint_checks] perform lints prior to macro expansion
+ #1 [hir_crate] getting the crate HIR
+ end of query stack
+ stack backtrace:
+ stack backtrace:
+    0:     0x7f64b76a4231 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h6dea4e7324a59a82
+    1:     0x7f64b770d258 - core::fmt::write::hfd96dfa72d0a2725
+    2:     0x7f64b7698711 - std::io::Write::write_fmt::h2e57173eeb62de35
+    3:     0x7f64b76a4041 - std::sys_common::backtrace::print::he2c4eabdf3f324b5
+    4:     0x7f64b76a71da - std::panicking::default_hook::{{closure}}::h5ac93fa12602a939
+    5:     0x7f64b76a6ec3 - std::panicking::default_hook::h744089070e358704
+    6:     0x7f64b83f580b - rustc_driver_impl[83116b0749ee6832]::install_ice_hook::{closure#0}
+    7:     0x7f64b76a791e - std::panicking::rust_panic_with_hook::hfa2b2d0711ad450a
+    8:     0x7f64b76a7679 - std::panicking::begin_panic_handler::{{closure}}::h5f4d8dedc2462c74
+    9:     0x7f64b76a46f6 - std::sys_common::backtrace::__rust_end_short_backtrace::h03e91b6dad2b248d
+   10:     0x7f64b76a7387 - rust_begin_unwind
+   11:     0x7f64b765b113 - core::panicking::panic_nounwind_fmt::h7ce9587ac45885db
+   12:     0x7f64b765b1b7 - core::panicking::panic_nounwind::h74409aad26871769
+   13:     0x7f64b765b343 - core::panicking::panic_cannot_unwind::h5acd21378cd039d4
+   14:     0x7f64b84958cd - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_expr
+   15:     0x7f64b8496980 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_stmt
+   16:     0x7f64b844fbdc - rustc_ast[5f32a37bf33b9536]::visit::walk_block::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
+   17:     0x7f64b84971c5 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_block
+   18:     0x7f64b849daa0 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_fn
+   19:     0x7f64b8456c9c - rustc_ast[5f32a37bf33b9536]::visit::walk_item::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
+   20:     0x7f64b8495bac - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_item
+   21:     0x7f64b844fd0c - rustc_ast[5f32a37bf33b9536]::visit::walk_crate::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
+   22:     0x7f64b8485de1 - rustc_lint[63ba0b95152488e9]::early::check_ast_node::<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass, (&rustc_ast[5f32a37bf33b9536]::ast::Crate, &[rustc_ast[5f32a37bf33b9536]::ast::Attribute])>
+   23:     0x7f64b84eb1eb - rustc_interface[d667fd10b95d5f1c]::passes::early_lint_checks
+   24:     0x7f64b9e38ce5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>
+   25:     0x7f64b9fa03c0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   26:     0x7f64ba11c68f - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   27:     0x7f64ba00da2d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::get_query_non_incr::__rust_end_short_backtrace
+   28:     0x7f64b9255930 - rustc_ast_lowering[2dcdb8df9e5d1db7]::lower_to_hir
+   29:     0x7f64b9e50015 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
+   30:     0x7f64b9f73240 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   31:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   32:     0x7f64ba00dafa - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::get_query_non_incr::__rust_end_short_backtrace
+   33:     0x7f64bb14d014 - <rustc_middle[f3ae9a0a99208368]::hir::provide::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
+   34:     0x7f64b9e502ae - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>
+   35:     0x7f64b9f040ac - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
+   36:     0x7f64ba1b4251 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   37:     0x7f64ba0ed246 - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::get_query_non_incr::__rust_end_short_backtrace
+   38:     0x7f64bb16e080 - rustc_middle[f3ae9a0a99208368]::query::plumbing::query_get_at::<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>>
+   39:     0x7f64bb1509db - <rustc_middle[f3ae9a0a99208368]::hir::map::Map>::get_module
+   40:     0x7f64bb1559ec - rustc_middle[f3ae9a0a99208368]::hir::map::hir_crate_items
+   41:     0x7f64b9e344b5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
+   42:     0x7f64b9f6fbb0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   43:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   44:     0x7f64ba0a0b6a - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::get_query_non_incr::__rust_end_short_backtrace
+   45:     0x7f64b96357b0 - rustc_passes[efdb855c787defeb]::hir_id_validator::check_crate
+   46:     0x7f64b84f04cf - rustc_interface[d667fd10b95d5f1c]::passes::analysis
+   47:     0x7f64b9e4ee05 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>
+   48:     0x7f64ba0ebbe0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
+   49:     0x7f64ba12135e - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
+   50:     0x7f64ba0a2b9d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
+   51:     0x7f64b83c2e4d - <rustc_middle[f3ae9a0a99208368]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   52:     0x7f64b83d3b08 - <rustc_interface[d667fd10b95d5f1c]::interface::Compiler>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}, core[ef4d13f605d7c29c]::result::Result<core[ef4d13f605d7c29c]::option::Option<rustc_interface[d667fd10b95d5f1c]::queries::Linker>, rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   53:     0x7f64b83a2d58 - rustc_span[90e41021240f79d4]::set_source_map::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
+   54:     0x7f64b8388d91 - <scoped_tls[c2bdb5f8b9cb5d8d]::ScopedKey<rustc_span[90e41021240f79d4]::SessionGlobals>>::set::<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   55:     0x7f64b841c926 - std[579a62aacf84a528]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
+   56:     0x7f64b8419bb6 - std[579a62aacf84a528]::panicking::try::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, core[ef4d13f605d7c29c]::panic::unwind_safe::AssertUnwindSafe<<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
+   57:     0x7f64b83a13af - <<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
+   58:     0x7f64b76b417d - std::sys::unix::thread::Thread::new::thread_start::h09e0297455ccbbb3
+   59:     0x7f64b744eb43 - <unknown>
+   60:     0x7f64b74e0a00 - <unknown>
+   61:                0x0 - <unknown>
+ error: the compiler unexpectedly panicked. this is a bug.
+ 
+ query stack during panic:
+ query stack during panic:
+ #0 [early_lint_checks] perform lints prior to macro expansion
+ #1 [hir_crate] getting the crate HIR
+ end of query stack
+ thread caused non-unwinding panic. aborting.


The actual stderr differed from the expected stderr.
The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/treat-err-as-bug/panic-causes-oom-112708/panic-causes-oom-112708.stderr
To only update this specific test, also pass `--test-args treat-err-as-bug/panic-causes-oom-112708.rs`

error: 1 errors occurred comparing output.
error: 1 errors occurred comparing output.
status: signal: 6 (SIGABRT) (core dumped)
command: RUST_BACKTRACE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/treat-err-as-bug/panic-causes-oom-112708" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/treat-err-as-bug/panic-causes-oom-112708/auxiliary" "-Ztreat-err-as-bug"
stdout: none
--- stderr -------------------------------
error: denote infinite loops with `loop { ... }`
##[error]  --> fake-test-src-base/treat-err-as-bug/panic-causes-oom-112708.rs:10:5
LL |     while true {}
   |     ^^^^^^^^^^ help: use `loop`
   |
note: the lint level is defined here
note: the lint level is defined here
  --> fake-test-src-base/treat-err-as-bug/panic-causes-oom-112708.rs:9:12
   |
LL |     #[deny(while_true)]
   |            ^^^^^^^^^^

thread 'rustc' panicked at 'aborting due to `-Z treat-err-as-bug=1`', compiler/rustc_errors/src/lib.rs:1711:30

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md
note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.72.0-nightly (d008ef9e9 2023-06-27) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z treat-err-as-bug
query stack during panic:
query stack during panic:
#0 [early_lint_checks] perform lints prior to macro expansion
#1 [hir_crate] getting the crate HIR
end of query stack
thread 'rustc' panicked at 'Found a `push` without a `pop`.', compiler/rustc_lint/src/levels.rs:516:9
stack backtrace:
   0:     0x7f64b76a4231 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h6dea4e7324a59a82
   1:     0x7f64b770d258 - core::fmt::write::hfd96dfa72d0a2725
   2:     0x7f64b7698711 - std::io::Write::write_fmt::h2e57173eeb62de35
   3:     0x7f64b76a4041 - std::sys_common::backtrace::print::he2c4eabdf3f324b5
   4:     0x7f64b76a71da - std::panicking::default_hook::{{closure}}::h5ac93fa12602a939
   5:     0x7f64b76a6ec3 - std::panicking::default_hook::h744089070e358704
   6:     0x7f64b83f580b - rustc_driver_impl[83116b0749ee6832]::install_ice_hook::{closure#0}
   7:     0x7f64b76a791e - std::panicking::rust_panic_with_hook::hfa2b2d0711ad450a
   8:     0x7f64b76a7679 - std::panicking::begin_panic_handler::{{closure}}::h5f4d8dedc2462c74
   9:     0x7f64b76a46f6 - std::sys_common::backtrace::__rust_end_short_backtrace::h03e91b6dad2b248d
  10:     0x7f64b76a7387 - rust_begin_unwind
  11:     0x7f64b765b0d3 - core::panicking::panic_fmt::h469794c234f7c718
  12:     0x7f64baaf75ef - <rustc_lint[63ba0b95152488e9]::levels::BuilderPush as core[ef4d13f605d7c29c]::ops::drop::Drop>::drop
  13:     0x7f64b84958b5 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_expr
  14:     0x7f64b8496980 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_stmt
  15:     0x7f64b844fbdc - rustc_ast[5f32a37bf33b9536]::visit::walk_block::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
  16:     0x7f64b84971c5 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_block
  17:     0x7f64b849daa0 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_fn
  18:     0x7f64b8456c9c - rustc_ast[5f32a37bf33b9536]::visit::walk_item::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
  19:     0x7f64b8495bac - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_item
  20:     0x7f64b844fd0c - rustc_ast[5f32a37bf33b9536]::visit::walk_crate::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
  21:     0x7f64b8485de1 - rustc_lint[63ba0b95152488e9]::early::check_ast_node::<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass, (&rustc_ast[5f32a37bf33b9536]::ast::Crate, &[rustc_ast[5f32a37bf33b9536]::ast::Attribute])>
  22:     0x7f64b84eb1eb - rustc_interface[d667fd10b95d5f1c]::passes::early_lint_checks
  23:     0x7f64b9e38ce5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>
  24:     0x7f64b9fa03c0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  25:     0x7f64ba11c68f - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  26:     0x7f64ba00da2d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::get_query_non_incr::__rust_end_short_backtrace
  27:     0x7f64b9255930 - rustc_ast_lowering[2dcdb8df9e5d1db7]::lower_to_hir
  28:     0x7f64b9e50015 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
  29:     0x7f64b9f73240 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  30:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  31:     0x7f64ba00dafa - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::get_query_non_incr::__rust_end_short_backtrace
  32:     0x7f64bb14d014 - <rustc_middle[f3ae9a0a99208368]::hir::provide::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
  33:     0x7f64b9e502ae - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>
  34:     0x7f64b9f040ac - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
  35:     0x7f64ba1b4251 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  36:     0x7f64ba0ed246 - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::get_query_non_incr::__rust_end_short_backtrace
  37:     0x7f64bb16e080 - rustc_middle[f3ae9a0a99208368]::query::plumbing::query_get_at::<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>>
  38:     0x7f64bb1509db - <rustc_middle[f3ae9a0a99208368]::hir::map::Map>::get_module
  39:     0x7f64bb1559ec - rustc_middle[f3ae9a0a99208368]::hir::map::hir_crate_items
  40:     0x7f64b9e344b5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
  41:     0x7f64b9f6fbb0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  42:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  43:     0x7f64ba0a0b6a - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::get_query_non_incr::__rust_end_short_backtrace
  44:     0x7f64b96357b0 - rustc_passes[efdb855c787defeb]::hir_id_validator::check_crate
  45:     0x7f64b84f04cf - rustc_interface[d667fd10b95d5f1c]::passes::analysis
  46:     0x7f64b9e4ee05 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>
  47:     0x7f64ba0ebbe0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  48:     0x7f64ba12135e - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  49:     0x7f64ba0a2b9d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  50:     0x7f64b83c2e4d - <rustc_middle[f3ae9a0a99208368]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  51:     0x7f64b83d3b08 - <rustc_interface[d667fd10b95d5f1c]::interface::Compiler>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}, core[ef4d13f605d7c29c]::result::Result<core[ef4d13f605d7c29c]::option::Option<rustc_interface[d667fd10b95d5f1c]::queries::Linker>, rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  52:     0x7f64b83a2d58 - rustc_span[90e41021240f79d4]::set_source_map::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  53:     0x7f64b8388d91 - <scoped_tls[c2bdb5f8b9cb5d8d]::ScopedKey<rustc_span[90e41021240f79d4]::SessionGlobals>>::set::<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  54:     0x7f64b841c926 - std[579a62aacf84a528]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  55:     0x7f64b8419bb6 - std[579a62aacf84a528]::panicking::try::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, core[ef4d13f605d7c29c]::panic::unwind_safe::AssertUnwindSafe<<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  56:     0x7f64b83a13af - <<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  57:     0x7f64b76b417d - std::sys::unix::thread::Thread::new::thread_start::h09e0297455ccbbb3
  58:     0x7f64b744eb43 - <unknown>
  59:     0x7f64b74e0a00 - <unknown>
  60:                0x0 - <unknown>
error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md


note: rustc 1.72.0-nightly (d008ef9e9 2023-06-27) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z treat-err-as-bug
query stack during panic:
query stack during panic:
#0 [early_lint_checks] perform lints prior to macro expansion
#1 [hir_crate] getting the crate HIR
thread 'rustc' panicked at 'panic in a function that cannot unwind', library/core/src/panicking.rs:126:5
stack backtrace:
stack backtrace:
   0:     0x7f64b76a4231 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h6dea4e7324a59a82
   1:     0x7f64b770d258 - core::fmt::write::hfd96dfa72d0a2725
   2:     0x7f64b7698711 - std::io::Write::write_fmt::h2e57173eeb62de35
   3:     0x7f64b76a4041 - std::sys_common::backtrace::print::he2c4eabdf3f324b5
   4:     0x7f64b76a71da - std::panicking::default_hook::{{closure}}::h5ac93fa12602a939
   5:     0x7f64b76a6ec3 - std::panicking::default_hook::h744089070e358704
   6:     0x7f64b83f580b - rustc_driver_impl[83116b0749ee6832]::install_ice_hook::{closure#0}
   7:     0x7f64b76a791e - std::panicking::rust_panic_with_hook::hfa2b2d0711ad450a
   8:     0x7f64b76a7679 - std::panicking::begin_panic_handler::{{closure}}::h5f4d8dedc2462c74
   9:     0x7f64b76a46f6 - std::sys_common::backtrace::__rust_end_short_backtrace::h03e91b6dad2b248d
  10:     0x7f64b76a7387 - rust_begin_unwind
  11:     0x7f64b765b113 - core::panicking::panic_nounwind_fmt::h7ce9587ac45885db
  12:     0x7f64b765b1b7 - core::panicking::panic_nounwind::h74409aad26871769
  13:     0x7f64b765b343 - core::panicking::panic_cannot_unwind::h5acd21378cd039d4
  14:     0x7f64b84958cd - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_expr
  15:     0x7f64b8496980 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_stmt
  16:     0x7f64b844fbdc - rustc_ast[5f32a37bf33b9536]::visit::walk_block::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
  17:     0x7f64b84971c5 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_block
  18:     0x7f64b849daa0 - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_fn
  19:     0x7f64b8456c9c - rustc_ast[5f32a37bf33b9536]::visit::walk_item::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
  20:     0x7f64b8495bac - <rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass> as rustc_ast[5f32a37bf33b9536]::visit::Visitor>::visit_item
  21:     0x7f64b844fd0c - rustc_ast[5f32a37bf33b9536]::visit::walk_crate::<rustc_lint[63ba0b95152488e9]::early::EarlyContextAndPass<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass>>
  22:     0x7f64b8485de1 - rustc_lint[63ba0b95152488e9]::early::check_ast_node::<rustc_lint[63ba0b95152488e9]::BuiltinCombinedEarlyLintPass, (&rustc_ast[5f32a37bf33b9536]::ast::Crate, &[rustc_ast[5f32a37bf33b9536]::ast::Attribute])>
  23:     0x7f64b84eb1eb - rustc_interface[d667fd10b95d5f1c]::passes::early_lint_checks
  24:     0x7f64b9e38ce5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>
  25:     0x7f64b9fa03c0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  26:     0x7f64ba11c68f - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  27:     0x7f64ba00da2d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::early_lint_checks::get_query_non_incr::__rust_end_short_backtrace
  28:     0x7f64b9255930 - rustc_ast_lowering[2dcdb8df9e5d1db7]::lower_to_hir
  29:     0x7f64b9e50015 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
  30:     0x7f64b9f73240 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  31:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  32:     0x7f64ba00dafa - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate::get_query_non_incr::__rust_end_short_backtrace
  33:     0x7f64bb14d014 - <rustc_middle[f3ae9a0a99208368]::hir::provide::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
  34:     0x7f64b9e502ae - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>
  35:     0x7f64b9f040ac - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, rustc_hir[f89174727a4d4899]::hir_id::OwnerId)>>::call_once
  36:     0x7f64ba1b4251 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  37:     0x7f64ba0ed246 - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_owner::get_query_non_incr::__rust_end_short_backtrace
  38:     0x7f64bb16e080 - rustc_middle[f3ae9a0a99208368]::query::plumbing::query_get_at::<rustc_query_system[f1667e0737e8102b]::query::caches::VecCache<rustc_hir[f89174727a4d4899]::hir_id::OwnerId, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 16usize]>>>
  39:     0x7f64bb1509db - <rustc_middle[f3ae9a0a99208368]::hir::map::Map>::get_module
  40:     0x7f64bb1559ec - rustc_middle[f3ae9a0a99208368]::hir::map::hir_crate_items
  41:     0x7f64b9e344b5 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>
  42:     0x7f64b9f6fbb0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  43:     0x7f64ba124523 - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  44:     0x7f64ba0a0b6a - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::hir_crate_items::get_query_non_incr::__rust_end_short_backtrace
  45:     0x7f64b96357b0 - rustc_passes[efdb855c787defeb]::hir_id_validator::check_crate
  46:     0x7f64b84f04cf - rustc_interface[d667fd10b95d5f1c]::passes::analysis
  47:     0x7f64b9e4ee05 - rustc_query_impl[8dba2f25e24dcd7a]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>
  48:     0x7f64ba0ebbe0 - <rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::dynamic_query::{closure#2} as core[ef4d13f605d7c29c]::ops::function::FnOnce<(rustc_middle[f3ae9a0a99208368]::ty::context::TyCtxt, ())>>::call_once
  49:     0x7f64ba12135e - rustc_query_system[f1667e0737e8102b]::query::plumbing::try_execute_query::<rustc_query_impl[8dba2f25e24dcd7a]::DynamicConfig<rustc_query_system[f1667e0737e8102b]::query::caches::SingleCache<rustc_middle[f3ae9a0a99208368]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[8dba2f25e24dcd7a]::plumbing::QueryCtxt, false>
  50:     0x7f64ba0a2b9d - rustc_query_impl[8dba2f25e24dcd7a]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  51:     0x7f64b83c2e4d - <rustc_middle[f3ae9a0a99208368]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  52:     0x7f64b83d3b08 - <rustc_interface[d667fd10b95d5f1c]::interface::Compiler>::enter::<rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}::{closure#2}, core[ef4d13f605d7c29c]::result::Result<core[ef4d13f605d7c29c]::option::Option<rustc_interface[d667fd10b95d5f1c]::queries::Linker>, rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  53:     0x7f64b83a2d58 - rustc_span[90e41021240f79d4]::set_source_map::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  54:     0x7f64b8388d91 - <scoped_tls[c2bdb5f8b9cb5d8d]::ScopedKey<rustc_span[90e41021240f79d4]::SessionGlobals>>::set::<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  55:     0x7f64b841c926 - std[579a62aacf84a528]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>
  56:     0x7f64b8419bb6 - std[579a62aacf84a528]::panicking::try::<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, core[ef4d13f605d7c29c]::panic::unwind_safe::AssertUnwindSafe<<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  57:     0x7f64b83a13af - <<std[579a62aacf84a528]::thread::Builder>::spawn_unchecked_<rustc_interface[d667fd10b95d5f1c]::util::run_in_thread_pool_with_globals<rustc_interface[d667fd10b95d5f1c]::interface::run_compiler<core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>, rustc_driver_impl[83116b0749ee6832]::run_compiler::{closure#1}>::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[ef4d13f605d7c29c]::result::Result<(), rustc_span[90e41021240f79d4]::ErrorGuaranteed>>::{closure#1} as core[ef4d13f605d7c29c]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  58:     0x7f64b76b417d - std::sys::unix::thread::Thread::new::thread_start::h09e0297455ccbbb3
  59:     0x7f64b744eb43 - <unknown>
  60:     0x7f64b74e0a00 - <unknown>
  61:                0x0 - <unknown>
error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md


note: rustc 1.72.0-nightly (d008ef9e9 2023-06-27) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z treat-err-as-bug
query stack during panic:
query stack during panic:
#0 [early_lint_checks] perform lints prior to macro expansion
#1 [hir_crate] getting the crate HIR
thread caused non-unwinding panic. aborting.
------------------------------------------


```

</details>

--- a/compiler/rustc_interface/src/interface.rs
+++ b/compiler/rustc_interface/src/interface.rs
@@ -349,7 +349,12 @@ pub fn try_print_query_stack(handler: &Handler, num_frames: Option<usize>) {
     // state if it was responsible for triggering the panic.
     let i = ty::tls::with_context_opt(|icx| {
         if let Some(icx) = icx {
-            print_query_stack(QueryCtxt::new(icx.tcx), icx.query, handler, num_frames)
+            ty::print::with_no_queries!(print_query_stack(
+                QueryCtxt::new(icx.tcx),
+                icx.query,
+                handler,
+                num_frames
+            ))
         } else {
             0
         }

--- a/compiler/rustc_middle/src/ty/print/pretty.rs
+++ b/compiler/rustc_middle/src/ty/print/pretty.rs
@@ -94,6 +94,10 @@ macro_rules! define_helper {
                     $tl.with(|c| c.set(self.0))
                 }
             }
+
+            pub fn $name() -> bool {
+                $tl.with(|c| c.get())
+            }
         )+
     }
 }
@@ -676,7 +680,7 @@ pub trait PrettyPrinter<'tcx>:
                 p!(")")
             }
             ty::FnDef(def_id, substs) => {
-                if NO_QUERIES.with(|q| q.get()) {
+                if with_no_queries() {
                     p!(print_def_path(def_id, substs));
                 } else {
                     let sig = self.tcx().fn_sig(def_id).subst(self.tcx(), substs);
@@ -732,7 +736,7 @@ pub trait PrettyPrinter<'tcx>:
                 p!(print_def_path(def_id, &[]));
             }
             ty::Alias(ty::Projection | ty::Inherent | ty::Weak, ref data) => {
-                if !(self.should_print_verbose() || NO_QUERIES.with(|q| q.get()))
+                if !(self.should_print_verbose() || with_no_queries())
                     && self.tcx().is_impl_trait_in_trait(data.def_id)
                 {
                     return self.pretty_print_opaque_impl_type(data.def_id, data.substs);
@@ -779,7 +783,7 @@ pub trait PrettyPrinter<'tcx>:
                         return Ok(self);
                     }
                     _ => {
-                        if NO_QUERIES.with(|q| q.get()) {
+                        if with_no_queries() {
                             p!(print_def_path(def_id, &[]));
                             return Ok(self);
                         } else {
@@ -1746,7 +1750,8 @@ impl DerefMut for FmtPrinter<'_, '_> {
 
 impl<'a, 'tcx> FmtPrinter<'a, 'tcx> {
     pub fn new(tcx: TyCtxt<'tcx>, ns: Namespace) -> Self {
-        Self::new_with_limit(tcx, ns, tcx.type_length_limit())
+        let limit = if with_no_queries() { Limit::new(1048576) } else { tcx.type_length_limit() };
+        Self::new_with_limit(tcx, ns, limit)
     }
 
     pub fn new_with_limit(tcx: TyCtxt<'tcx>, ns: Namespace, type_length_limit: Limit) -> Self {

--- a/compiler/rustc_middle/src/ty/print/pretty.rs
+++ b/compiler/rustc_middle/src/ty/print/pretty.rs
@@ -3004,7 +3004,7 @@ fn for_each_def(tcx: TyCtxt<'_>, mut collect_fn: impl for<'b> FnMut(&'b Ident, N
 ///
 /// The implementation uses similar import discovery logic to that of 'use' suggestions.
 ///
-/// See also [`DelayDm`](rustc_error_messages::DelayDm) and [`with_no_trimmed_paths`].
+/// See also [`DelayDm`](rustc_error_messages::DelayDm) and [`with_no_trimmed_paths!`].
 fn trimmed_def_paths(tcx: TyCtxt<'_>, (): ()) -> FxHashMap<DefId, Symbol> {
     let mut map: FxHashMap<DefId, Symbol> = FxHashMap::default();
 

--- a/compiler/rustc_query_impl/src/plumbing.rs
+++ b/compiler/rustc_query_impl/src/plumbing.rs
@@ -16,7 +16,7 @@ use rustc_middle::query::on_disk_cache::AbsoluteBytePos;
 use rustc_middle::query::on_disk_cache::{CacheDecoder, CacheEncoder, EncodedDepNodeIndex};
 use rustc_middle::query::Key;
 use rustc_middle::ty::tls::{self, ImplicitCtxt};
-use rustc_middle::ty::{self, TyCtxt};
+use rustc_middle::ty::{self, print::with_no_queries, TyCtxt};
 use rustc_query_system::dep_graph::{DepNodeParams, HasDepContext};
 use rustc_query_system::ich::StableHashingContext;
 use rustc_query_system::query::{
@@ -312,7 +312,7 @@ pub(crate) fn create_query_frame<
     );
     let description =
         if tcx.sess.verbose() { format!("{description} [{name:?}]") } else { description };
-    let span = if kind == dep_graph::DepKind::def_span {
+    let span = if kind == dep_graph::DepKind::def_span || with_no_queries() {
         // The `def_span` query is used to calculate `default_span`,
         // so exit to avoid infinite recursion.
         None
@@ -320,7 +320,7 @@ pub(crate) fn create_query_frame<
         Some(key.default_span(tcx))
     };
     let def_id = key.key_as_def_id();
-    let def_kind = if kind == dep_graph::DepKind::opt_def_kind {
+    let def_kind = if kind == dep_graph::DepKind::opt_def_kind || with_no_queries() {
         // Try to avoid infinite recursion.
         None
     } else {

--- a/tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs
+++ b/tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs
@@ -1,0 +1,11 @@
+// compile-flags: -Ztreat-err-as-bug
+// dont-check-failure-status
+// error-pattern: aborting due to `-Z treat-err-as-bug=1`
+// normalize-stderr-test "note: .*\n\n" -> ""
+// normalize-stderr-test "thread 'rustc' panicked.*\n" -> ""
+// rustc-env:RUST_BACKTRACE=0
+
+fn main() {
+    #[deny(while_true)]
+    while true {}
+}

--- a/tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs
+++ b/tests/ui/treat-err-as-bug/panic-causes-oom-112708.rs
@@ -1,8 +1,7 @@
 // compile-flags: -Ztreat-err-as-bug
 // dont-check-failure-status
 // error-pattern: aborting due to `-Z treat-err-as-bug=1`
-// normalize-stderr-test "note: .*\n\n" -> ""
-// normalize-stderr-test "thread 'rustc' panicked.*\n" -> ""
+// dont-check-compiler-stderr
 // rustc-env:RUST_BACKTRACE=0
 
 fn main() {

--- a/tests/ui/treat-err-as-bug/panic-causes-oom-112708.stderr
+++ b/tests/ui/treat-err-as-bug/panic-causes-oom-112708.stderr
@@ -1,0 +1,16 @@
+error: denote infinite loops with `loop { ... }`
+  --> $DIR/panic-causes-oom-112708.rs:10:5
+   |
+LL |     while true {}
+   |     ^^^^^^^^^^ help: use `loop`
+   |
+note: the lint level is defined here
+  --> $DIR/panic-causes-oom-112708.rs:9:12
+   |
+LL |     #[deny(while_true)]
+   |            ^^^^^^^^^^
+
+error: the compiler unexpectedly panicked. this is a bug.
+
+query stack during panic:
+thread panicked while processing panic. aborting.

--- a/tests/ui/treat-err-as-bug/panic-causes-oom-112708.stderr
+++ b/tests/ui/treat-err-as-bug/panic-causes-oom-112708.stderr
@@ -1,16 +1,32 @@
 error: denote infinite loops with `loop { ... }`
-  --> $DIR/panic-causes-oom-112708.rs:10:5
+  --> $DIR/panic-causes-oom-112708.rs:13:5
    |
 LL |     while true {}
    |     ^^^^^^^^^^ help: use `loop`
    |
 note: the lint level is defined here
-  --> $DIR/panic-causes-oom-112708.rs:9:12
+  --> $DIR/panic-causes-oom-112708.rs:12:12
    |
 LL |     #[deny(while_true)]
    |            ^^^^^^^^^^
 
+
+query stack during panic:
+#0 [early_lint_checks] perform lints prior to macro expansion
+#1 [hir_crate] getting the crate HIR
+end of query stack
+
 error: the compiler unexpectedly panicked. this is a bug.
 
 query stack during panic:
-thread panicked while processing panic. aborting.
+#0 [early_lint_checks] perform lints prior to macro expansion
+#1 [hir_crate] getting the crate HIR
+end of query stack
+
+error: the compiler unexpectedly panicked. this is a bug.
+
+query stack during panic:
+#0 [early_lint_checks] perform lints prior to macro expansion
+#1 [hir_crate] getting the crate HIR
+end of query stack
+thread caused non-unwinding panic. aborting.

